hooks/python-lint-pre-commit

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+# Copyright 2018 The Pontem Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Performs pylint on code being committed.
+#
+echo "Starting pre-commit"
+# get all files
+files=`git diff --name-only --cached '*.py'`
+if [ ${#files[@]} -eq 0 ]; then
+  echo "No python files to lint"
+else
+  declare -A root_module_folders
+
+  for f in $files
+  do
+    project_root=`echo ${f} | cut -d '/' -f1-1`
```

Comment:
I still don't follow why you need all this path manipulation pre-commits are run relative to the repo's root IIRC and gitdiff outputs files in this format so I'm unsure why all this is needed. Is it related to helping pylint resolve imports?

Reply:
It is because there is no __init__.py at the project level. Pylint gives the following error:

F:  1, 0: error while code parsing: Unable to load file CloudSQLReplicator/__init__.py:
[Errno 2] No such file or directory: 'CloudSQLReplicator/__init__.py' (parse-error)